docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,595 @@
+    kind: RouteAdvertisements
+    metadata:
+      name: blue
+    spec:
+      # advertise routes to target VRF vpn-blue
+      targetVRF: auto
+      # nodeSelector must be empty, since we don't support a network in a overlay and no-overlay hybrid mode.
+      nodeSelector: {}
+      frrConfigurationSelector:
+        network: vpn-blue
+      networkSelectors:
+      - networkSelectionType: ClusterUserDefineNetwork
+        clusterUserDefinedNetworkSelector:
+          networkSelector:
+            matchLabels:
+              # Select one CUDN to ensure the cross UDN isolation
+              advertise: blue
+      advertisements:
+      - PodNetwork
+    ```
+
+1. OVN-Kubernetes will generate a following FRRConfiguration for each node.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: route-generated-blue
+      namespace: frr-k8s-system
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          - address: 182.18.0.5
+            disableMP: true
+            asn: 64512
+            toAdvertise:
+              allowed:
+                prefixes:
+                # the node subnet of this node
+                - 10.10.1.0/24
+          vrf: blue
+          prefixes:
+          - 10.0.1.0/24
+    ```
+
+### Deployment Consideration
+
+#### BGP Topology
+
+The deployment mode will be decided by the FRRConfiguration CR that is selected
+for the no-overlay network. There will be no new field added to the
+FRRConfiguration.
+
+##### Full-Mesh iBGP across the Cluster
+
+The FRR instance on each node maintains a full mesh BGP peer relationship with
+all other nodes across the cluster. In this mode, users can enable the
+no-overlay mode without relying on external BGP routes.
+
+##### External Route Reflectors for Larger Clusters
+
+In a large cluster, a full-mesh BGP setup leads to more CPU and memory
+consumption on the nodes. Instead of every node peering with every other node,
+nodes peer only with external BGP route reflectors. This significantly reduces
+the number of BGP sessions each individual node needs to maintain, improving
+scalability.
+
+##### Internal Route Reflectors
+
+A FRR instances can also function as a BGP route reflectors, technically we can
+use internal route reflectors to build a BGP network across the cluster.
+However, the frr-k8s which is used for integrating OVN-K with FRR does not
+currently support the configuration of FRR as a route reflector, therefore this
+deployment mode will not be supported.
+
+### Feature Compatibility
+
+#### Multiple External Gateways (MEG)
+
+The same as the route advertisement feature.
+
+#### Egress IP
+
+Full support
+
+#### Services
+
+The same as the route advertisement feature.
+
+#### Egress Service
+
+Full support.
+
+#### Egress Firewall
+
+Full support.
+
+#### Egress QoS
+
+Full Support.
+
+#### Network Policy/ANP
+
+Full Support.
+
+#### IPsec
+
+Not Support.
```

Comment:
so on ipsec cluster, there will `not supported` message shown when applying the no overlay CR?